### DISCUSSION TITLE
Comparison of Move with NO_MOVE

### DISCUSSION THREAD

**Discussion by Orbital-Web** [Disservin/chess-library#21] [Q&A] [answered]
Hi, I noticed that I can no longer just compare `chess::Move move` with `chess::Move::NO_MOVE` using ==. Instead, I need to compare NO_MOVE against `move.move()`. Same goes for returning `chess::Move::NO_MOVE`, I now have to convert it to type `chess::Move` explicitly.

I believe this is because of the `explicit` keyword in `constexpr explicit Move(uint16_t move)` in line 435 (as of latest commit).
Is there a reason why `explicit` was added, and is it possible to remove it to allow implicit conversion between NO_MOVE and move?

**Comment by Disservin** [comment, marked as answer]
Thanks for pointing it out, ty! I added it back with <https://github.com/Disservin/chess-library/commit/f0c27d67de62f968ea7b6b9dd87bf84b4ed7ebd3>.

I also just changed the name of `MoveGenType::CAPTURE` to `MoveGenType::NOISY`, since it was actually returning all noisy moves and captures was a bit misleading... what's your feedback on this? Do you want a standalone function which returns capture moves and one which also includes noisy?

**Reply by Disservin** [reply]
Okay, sorry I changed the new noisy type actually back to `CAPTURE`. In the old version `CAPTURE` was actually also generating non capture promotion moves. This is now fixed properly, so that `CAPTURE` generates any capture move and `QUIET` all non capture moves.

**Reply by Orbital-Web** [reply]
I think having the noisy movegen is still quite useful (captures + promotions (possibly queening only)) for implementing quiescence search in chess engines. 
Having a way to check if a move is going to result in a check beforehand is also useful, though I can't think of an efficient approach to do that at the top of my head, besides playing and undoing the move.

**Reply by Disservin** [reply]
There are some efficient approachs but eh I dont really plan on implementing one right now. Regarding the noisy moves maybe ill add it lets see.
